pkg/testutils/exec.go

```diff
@@ -101,3 +103,35 @@ func StartService(h, service string) error {
 	}
 	return nil
 }
+
+// WaitForNodeUP waits until nodetool status report UN status for the given node.
+// The nodetool status CLI is executed on the same node.
+func WaitForNodeUP(h string, timeout time.Duration) error {
+	nodeIsReady := make(chan struct{})
+	done := make(chan struct{})
+	go func() {
+		defer close(nodeIsReady)
+		for {
+			select {
+			case <-done:
+				return
+			case <-time.After(time.Second):
```

Comment:
So we wait 1 second even before the first check? This could be problematic with small timeouts like 1 second.

Reply:
You are right. Fixed